src/object/srv_obj_migrate.c

```diff
@@ -2352,12 +2353,13 @@ migrate_cont_iter_cb(daos_handle_t ih, d_iov_t *key_iov,
 	if (rc1 != 0 || rc != 0)
 		D_GOTO(free, rc = rc ? rc : rc1);
 
-	D_DEBUG(DB_TRACE, "iter cont "DF_UUID"/%"PRIx64" finish.\n",
+	D_DEBUG(DB_REBUILD, "iter cont "DF_UUID"/%"PRIx64" finish.\n",
 		DP_UUID(cont_uuid), ih.cookie);
 
 	/* Snapshot fetch will yield the ULT, let's reprobe before delete  */
+	d_iov_set(&tmp_iov, cont_uuid, sizeof(uuid_t));
```

Comment:
I don't quite follow this change. Why is the 'key_iov' invalid now? Will dbtree change (while yielding) invalidate the 'key_iov'? Or the key was deleted by others while yielding?

Reply:
the key_iov.iov_buf is directly from the record by offset (see nv_rec_fetch), so if there are new insertion, the iov_buf will change anyway, so we need reset the key.

Reply:
Right, the key is from btr_record which could be moved around on insertion. What I don't quite follow is: Looks this ULT is iterating & deleting on this container tree (it may yield while iterating or iterating in sub-level object tree), and the same time, this same container tree is updated (new containers & objects being inserted) by other ULT? How these two ULTs cooperate?